app/src/main/java/org/p2p/wallet/home/addmoney/ui/AddMoneyDialog.kt

```diff
@@ -0,0 +1,154 @@
+import org.p2p.wallet.databinding.DialogAddMoneyBinding
+import org.p2p.wallet.home.addmoney.AddMoneyDialogContract
+import org.p2p.wallet.home.addmoney.model.AddMoneyItemType
+import org.p2p.wallet.moonpay.model.PaymentMethod
+import org.p2p.wallet.moonpay.ui.new.NewBuyFragment
+import org.p2p.wallet.receive.ReceiveFragmentFactory
+import org.p2p.wallet.striga.StrigaFragmentFactory
+import org.p2p.wallet.striga.user.model.StrigaUserStatusDestination
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.toPx
+import org.p2p.wallet.utils.viewbinding.viewBinding
+
+class AddMoneyDialog :
+    BaseMvpBottomSheet<AddMoneyDialogContract.View, AddMoneyDialogContract.Presenter>(R.layout.dialog_add_money),
+    AddMoneyDialogContract.View {
+
+    companion object {
+        fun show(fm: FragmentManager) {
+            val tag = AddMoneyDialog::javaClass.name
+            if (fm.findFragmentByTag(tag) != null) return
+            AddMoneyDialog().show(fm, tag)
+        }
+    }
+
+    private val binding: DialogAddMoneyBinding by viewBinding()
+    private val receiveFragmentFactory: ReceiveFragmentFactory by inject()
+    private val strigaFragmentFactory: StrigaFragmentFactory by inject()
+
+    override val presenter: AddMoneyDialogContract.Presenter by inject()
+
+    private val cellAdapter = CommonAnyCellAdapter(
+        mainCellDelegate(inflateListener = {
+            it.setOnClickAction { _, item ->
+                presenter.onItemClick(item.typedPayload())
+            }
+        })
+    )
+
+    override fun getTheme(): Int = R.style.WalletTheme_BottomSheet_RoundedSnow
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            recyclerViewCells.disableScrolling()
+            recyclerViewCells.attachAdapter(cellAdapter)
+            recyclerViewCells.addItemDecoration(offsetFinanceBlockDecoration(8.toPx()))
+        }
+    }
+
+    override fun showItemProgress(itemType: AddMoneyItemType, showProgress: Boolean) {
+        cellAdapter.updateItem<MainCellModel>(
+            predicate = { it is MainCellModel && it.payload == itemType },
+            transform = {
+                val rightSideCellModel = if (showProgress) {
+                    RightSideCellModel.Progress(
+                        indeterminateProgressTint = R.color.night
+                    )
+                } else {
+                    RightSideCellModel.IconWrapper(
+                        iconWrapper = IconWrapperCellModel.SingleIcon(
+                            icon = ImageViewCellModel(
+                                icon = DrawableContainer(R.drawable.ic_chevron_right),
+                                iconTint = R.color.icons_mountain
+                            )
+                        )
+                    )
+                }
+                it.copy(rightSideCellModel = rightSideCellModel)
```

Comment:
Looks really strange. 
Should we do this in presenter and just send a new list to adapter which should update the cell automatically?

Reply:
In my opinion, boilerplating in this case is unnecessary, as it's already doing the same thing but within adapter. I don't see a significant advantage in moving the logic to presenter + mapper, but okay, I'll do this

Reply:
@1sakov moved progress logic to the presenter, check please the new implementation